src/api/submit_order.js

```diff
@@ -0,0 +1,32 @@
+const { post } = require('request-promise')
+
+module.exports = async (efx, symbol, amount, price, gid, cid) => {
+  if (!(symbol && amount && price)) {
+    throw new Error('symbol, amount and price are required')
+  }
+
+  gid = gid || 1 // REVIEW: shall we make it unique as well?
+  cid = cid || new Date().getTime()
+
+  const signed = await efx.sign.order(symbol, amount, price)
+
+  // REVIEW: shall we use signed.ecSignature ?
+  // or refactor it to just signed.signature ?
+  // or even just return the signature? ( this would be problematic for tests )
+  const meta = { protocol: '0x', object: signed.signature }
```

Comment:
@prdn / @plutoegg I guess it's on the `object:` field that we will attach the signature?

Reply:
The object needs to contain the full order as well as the signature

i.e. `object: signedOrder`

signOrder contains all params, as well as an additional field ecSignature
```
const signedOrder = {
    expirationUnixTimestampSec: ...,
    feeRecipient: ...,
    maker: ...,
    ecSignature: {
       v: ...,
       r: ...,
       s: ...,
    },
}

```

Reply:
okie